integration_tests/webhook/test_webhook.py

```diff
@@ -41,12 +36,60 @@ def test_webhook(self):
                     break
             if channel_id is not None:
                 break
+        return channel_id
+
+    def test_webhook(self):
+        url = os.environ[SLACK_SDK_TEST_INCOMING_WEBHOOK_URL]
+        webhook = WebhookClient(url)
+        response = webhook.send(text="Hello!")
+        self.assertEqual(200, response.status_code)
+        self.assertEqual("ok", response.body)
 
-        history = client.conversations_history(channel=channel_id, limit=1)
+        token = os.environ[SLACK_SDK_TEST_BOT_TOKEN]
+        client = WebClient(token=token)
+        history = client.conversations_history(channel=self.__get_channel_id(), limit=1)
         self.assertIsNotNone(history)
         actual_text = history["messages"][0]["text"]
         self.assertEqual("Hello!", actual_text)
 
+    def test_with_unfurls_off(self):
+        url = os.environ[SLACK_SDK_TEST_INCOMING_WEBHOOK_URL]
+        token = os.environ[SLACK_SDK_TEST_BOT_TOKEN]
+        webhook = WebhookClient(url)
+        client = WebClient(token=token)
+        # send message that does not unfurl
+        response = webhook.send(
+            text="<https://imgs.xkcd.com/comics/desert_golfing_2x.png|Desert Golfing>",
+            unfurl_links=False,
+            unfurl_media=False,
+        )
+        self.assertEqual(200, response.status_code)
+        self.assertEqual("ok", response.body)
+        # wait in order to allow Slack API to edit message with attachments
+        time.sleep(2)
+        history = client.conversations_history(channel=self.__get_channel_id(), limit=1)
+        self.assertIsNotNone(history)
+        self.assertTrue("attachments" not in history["messages"][0])
+
+    def test_with_unfurls_on(self):
+        url = os.environ[SLACK_SDK_TEST_INCOMING_WEBHOOK_URL]
+        token = os.environ[SLACK_SDK_TEST_BOT_TOKEN]
+        webhook = WebhookClient(url)
+        client = WebClient(token=token)
+        # send message that does unfurl
+        response = webhook.send(
+            text="<https://imgs.xkcd.com/comics/red_spiders_small.jpg|Spiders>",
+            unfurl_links=True,
+            unfurl_media=True,
+        )
+        self.assertEqual(200, response.status_code)
+        self.assertEqual("ok", response.body)
+        # wait in order to allow Slack API to edit message with attachments
+        time.sleep(2)
+        history = client.conversations_history(channel=self.__get_channel_id(), limit=1)
+        self.assertIsNotNone(history)
+        self.assertTrue("attachments" in history["messages"][0])
```

Comment:
I am using the existence of `attachments` to test whether the link has unfurled, as Slack API will scan a message one posted, and then edit the message body with attachments.

Once a link has been unfurled in a channel, Slack API will not unfurl it a second time, even if the `unfurl_links` and `unfurl_media` are both set to `True` . If a user of the Slack client does this manually, Slack sends an ephemeral message with an option to unfurl anyway (see image). I don't think there's an equivalent in the response to the Incoming Hooks endpoint.

If that's true then this test is unreliable when run multiple times, giving a true positive on the first run and a false negative on the second run. It doesn't seem realistic to modify the text field between test runs to overcome this. Do you have any thoughts on how I might make the this test better?

_Example in the Slack client_
<img width="1148" alt="Screen Shot 2021-06-24 at 1 22 24 PM" src="https://user-images.githubusercontent.com/55667998/123327431-4122bd00-d4ef-11eb-9c71-8d56539d90f6.png">

_Webhook API response_
![Screen Shot 2021-06-24 at 1 33 53 PM](https://user-images.githubusercontent.com/55667998/123328755-d2defa00-d4f0-11eb-8104-c90275a952da.png)
_

Reply:
This is a good question. 

>Do you have any thoughts on how I might make the this test better?

As we run these integration tests manually, this limitation is okay so far. You can delete the same images in the same channel as necessary. How about adding some comments near the assertions (say, if this fails when you run the same test multiple times, you need to use a different URL or delete the already existing attachments in the channel).